compiler/ballerina-lang/src/main/java/io/ballerina/projects/CompilationOptions.java

```diff
@@ -218,6 +239,16 @@ CompilationOptions acceptTheirs(CompilationOptions theirOptions) {
         } else {
             compilationOptionsBuilder.setRemoteManagement(this.remoteManagement);
         }
+        if (theirOptions.optimizeCodegen != null) {
```

Comment:
Accessing static fields via instance is confusing. But I assume above fix should correct this as well

Reply:
Fixed in ca52755bff1c0ddaeb396ed614106173315fdaa7.